### DISCUSSION TITLE
Multi-value field does not refresh from database changes (was: How to reload a table in the UI)

### DISCUSSION THREAD

**Discussion by christophberger** [teableio/teable#593] [Q&A]
I connected DBeaver to the database and modified a few rows. In the Teable UI, the old data keeps showing. 

How can I refresh/reload the table to fetch the changes from the database?

___

Update: The tables seem to reload, but I changed some tags (a multi-select field), and the changes don't show in the UI. If I change tags in the UI instead, the change appears properly in the database. 

So the behavior seems to be specific to multi-select fields.
___

Update 2: The table is imported from CSV. The tags column is of DB type JSONB and of cellValueType string.

**Comment by tea-artist** [comment]
Teable is currently designed to control the entire database. some functionality may not work correctly if you modify the database directly instead of through teable API or UI
> https://help.teable.io/developer/database-connection#direct-database-writing

**Comment by christophberger** [comment]
Thanks for the explanation. Do I understand this correctly: A small change in the column of a record breaks the logic?

Background: 
I imported a CSV with multiple tags in one column. Naturally, the CSV encloses the comma-separated tags in quotes, to avoid messing with the CSV commas. So, in the database, the `tags` column (configured in the UI as a multi-select field) contains the tags as a single value:

```

["abstraction, leaky_abstraction, software_development"]
```

Teable takes this as a single tag:

<img width="325" alt="Screenshot 2024-05-08 at 08 08 46" src="https://github.com/teableio/teable/assets/3835890/5e32561e-0577-4ac6-bf74-8357be0ae0ae">

Now, when I change the database column to the correct format in DBeaver...

```
["abstraction", "leaky_abstraction", "software_development"]
```

...then the UI shows nothing:

<img width="352" alt="Screenshot 2024-05-08 at 08 10 38" src="https://github.com/teableio/teable/assets/3835890/488cf182-fad9-4773-b952-86a54fd1ea09">

I know that I should fix the CSV before importing, but this is not the point here. The point is that I cannot alter column values directly without breaking the UI.

Is this a bug? Shall I open an issue?
Or is this expected behavior, based on your advice that the database must not be altered?

If the latter applies, I cannot use Teable for my use case, which requires that other apps can interact with the database directly. (Which works perfectly fine with NocoDB, by the way.)

**Comment by tea-artist** [comment]
Yes, the database must not be altered, currently, any problems caused by direct modifications to the database will not be supported.
In terms of implementation, this is feasible, but many details need to be worked out to support direct modification of the database.

(by the way, thanks for providing the detailed scene, which will provide a good reference for our subsequent design

**Comment by tea-artist** [comment]
After our discussion, we feel that this function is of great practical value, and we decided to implement this capability in a future release, thanks for the feedback!
